django_s3_storage/storage.py

```diff
@@ -356,12 +364,12 @@ def exists(self, name):
         if name.endswith("/"):
             # This looks like a directory, but on S3 directories are virtual, so we need to see if the key starts
             # with this prefix.
-            results = self.s3_connection.list_objects_v2(
-                Bucket=self.settings.AWS_S3_BUCKET_NAME,
-                MaxKeys=1,
-                Prefix=self._get_key_name(name) + "/",  # Add the slash again, since _get_key_name removes it.
-            )
-            return "Contents" in results
```

Comment:
There's no need to wrap anything, or make a separate method. You can just put a try/except around the call to `list_objects_v2`, and catch `ClientError`.

Reply:
This makes the code more consistent

Reply:
I'm not sure it's *consistent* to catch a `ClientError`, raise an `OSError`, then catch the `OSError`. That just sounds like more code, and slower code to me. :stuck_out_tongue: 

The `_wrap_errors` decorator exists to convert a `ClientError` to an `OSError` in public-facing methods. Since the exception is not raised in public-facing methods, there's no point in using it.

I've merged the fix to master. Please give it a try, and confirm that it's working for you. I'll then issue a bugfix release ASAP.

Thanks for the report, and debugging the solution!